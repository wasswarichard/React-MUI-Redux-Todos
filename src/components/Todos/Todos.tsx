import React, {useState, useEffect} from 'react';
import Board from "../Board/Board";
import Typography from '@material-ui/core/Typography';
import TodosList from "../TodosList/TodosList";
import axios from "axios";
import config from "../../utils/config.json";
import store from "../../state/store/store";
import {todosAdded} from "../../state/actions/actions";
import {makeStyles} from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
    heading1: {
        marginLeft: "20%",
    },
    heading2: {
        marginLeft: "10%",
    },
}), { name: "MuiButtonBase-root" });

const Todos = () => {
    const styles = useStyles();
    const [todos , setTodos] = useState(null);
    const [statusChange, setStatusChange] = useState({});

    useEffect(() => {
        axios.get(`${config.apiUrl}`)
            .then(response => {
                response.data.map((todo : any) => todo.status = "TODO");
                store.dispatch(todosAdded(response.data));
                setTodos(response.data)
            });

    }, []);

    useEffect(() => {
        const newTodos : any = store.getState();
        setTodos(newTodos);
    }, [statusChange])

    return (
        <div>
            <Typography gutterBottom variant="h5" className={styles.heading1}>Tickets List</Typography>
            {todos && <TodosList todos={todos} setStatusChange={setStatusChange}/>}
            <Typography gutterBottom variant="h5" className={styles.heading2}>Board</Typography>
            {todos && <Board todos={todos} setStatusChange={setStatusChange}/>}
        </div>
    );
}
export default Todos;