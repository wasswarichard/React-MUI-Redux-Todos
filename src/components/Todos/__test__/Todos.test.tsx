import React from 'react';
import Todos from '../Todos';
import Enzyme, { shallow } from 'enzyme';
import { cleanup } from '@testing-library/react';
import Adapter from 'enzyme-adapter-react-16';
import TestRenderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

describe('<Todos>', () => {
  afterEach(cleanup);

  it('renders BoardColumn Component', () => {
    shallow(<Todos />);
  });

  it('should match snapshot', function () {
    const tree = TestRenderer.create(<Todos />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
