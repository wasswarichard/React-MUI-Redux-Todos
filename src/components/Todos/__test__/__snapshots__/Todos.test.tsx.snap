// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todos> should match snapshot 1`] = `
<div>
  <h5
    className="MuiTypography-root MuiButtonBase-root-heading1 MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Tickets List
  </h5>
  <h5
    className="MuiTypography-root MuiButtonBase-root-heading2 MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Board
  </h5>
</div>
`;
